dotCMS/src/main/java/com/dotmarketing/portlets/structure/action/EditFieldAction.java

```diff
@@ -114,71 +115,80 @@ public void processAction(ActionMapping mapping, ActionForm form, PortletConfig
+                                    LegacyFieldTypes.HIDDEN.implClass().getCanonicalName().equals(newField.typeName())) {
+                                field.setValues(fieldForm.getValues());
+                            }
+
+                            BeanUtils.copyProperties(fieldForm, field);
                         }
-                        
-                        BeanUtils.copyProperties(fieldForm, field);
                     }
-                }
-                if (Validator.validate(req, form, mapping)) {
-                    if (_saveField(form, req, res, user)) {
-                        _sendToReferral(req, res, referer);
-                        return;
+                    if (Validator.validate(req, form, mapping)) {
+                        if (_saveField(form, req, res, user)) {
+                            _sendToReferral(req, res, referer);
+                            return;
+                        }
                     }
+                } catch (Exception ae) {
+                    _handleException(ae, req);
+                    return;
                 }
-            } catch (Exception ae) {
-                _handleException(ae, req);
-                return;
             }
-        }
         /*
          * If we are deleting the field, run the delete action and return to the
          * list
          *
          */
-        else if ((cmd != null) && cmd.equals(Constants.DELETE)) {
-            try {
-                Logger.debug(this, "Calling Delete Method");
-                _deleteField(req);
-            } catch (Exception ae) {
-                _handleException(ae, req);
-                return;
+            else if ((cmd != null) && cmd.equals(Constants.DELETE)) {
+                try {
+                    Logger.debug(this, "Calling Delete Method");
+                    _deleteField(req);
+                } catch (Exception ae) {
+                    _handleException(ae, req);
+                    return;
+                }
+                _sendToReferral(req, res, referer);
+            } else if ((cmd != null) && cmd.equals("reorder")) {
+                try {
+                    Logger.debug(this, "Calling reorder Method");
+                    _reorderFields(form, req, res);
+                } catch (Exception ae) {
+                    _handleException(ae, req);
+                    return;
+                }
+                _sendToReferral(req, res, referer);
             }
-            _sendToReferral(req, res, referer);
-        } else if ((cmd != null) && cmd.equals("reorder")) {
+        } finally {
             try {
-                Logger.debug(this, "Calling reorder Method");
-                _reorderFields(form, req, res);
-            } catch (Exception ae) {
-                _handleException(ae, req);
-                return;
+                HibernateUtil.closeAndCommitTransaction();
+            } catch (DotHibernateException e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1166)